# This file is part of HiGFXback

REQUIRES="autotools-wrappers-am-build expat-build zlib-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15)
aclocal -I m4; autoconf; autoheader
./configure --enable-shared --with-system-expat --without-ensurepip --prefix=/

# build
make

# install
install -d $DESTDIR/bin
install python $DESTDIR/bin/python3
install -d $DESTDIR/include/python3
install -m 644 pyconfig.h $DESTDIR/include/python3
install -m 644 Include/abstract.h $DESTDIR/include/python3
install -m 644 Include/bltinmodule.h $DESTDIR/include/python3
install -m 644 Include/boolobject.h $DESTDIR/include/python3
install -m 644 Include/bytearrayobject.h $DESTDIR/include/python3
install -m 644 Include/bytesobject.h $DESTDIR/include/python3
install -m 644 Include/cellobject.h $DESTDIR/include/python3
install -m 644 Include/ceval.h $DESTDIR/include/python3
install -m 644 Include/classobject.h $DESTDIR/include/python3
install -m 644 Include/code.h $DESTDIR/include/python3
install -m 644 Include/codecs.h $DESTDIR/include/python3
install -m 644 Include/compile.h $DESTDIR/include/python3
install -m 644 Include/complexobject.h $DESTDIR/include/python3
install -m 644 Include/context.h $DESTDIR/include/python3
install -m 644 Include/descrobject.h $DESTDIR/include/python3
install -m 644 Include/dictobject.h $DESTDIR/include/python3
install -m 644 Include/dtoa.h $DESTDIR/include/python3
install -m 644 Include/enumobject.h $DESTDIR/include/python3
install -m 644 Include/eval.h $DESTDIR/include/python3
install -m 644 Include/fileobject.h $DESTDIR/include/python3
install -m 644 Include/fileutils.h $DESTDIR/include/python3
install -m 644 Include/floatobject.h $DESTDIR/include/python3
install -m 644 Include/frameobject.h $DESTDIR/include/python3
install -m 644 Include/funcobject.h $DESTDIR/include/python3
install -m 644 Include/genobject.h $DESTDIR/include/python3
install -m 644 Include/import.h $DESTDIR/include/python3
install -m 644 Include/intrcheck.h $DESTDIR/include/python3
install -m 644 Include/iterobject.h $DESTDIR/include/python3
install -m 644 Include/listobject.h $DESTDIR/include/python3
install -m 644 Include/longintrepr.h $DESTDIR/include/python3
install -m 644 Include/longobject.h $DESTDIR/include/python3
install -m 644 Include/memoryobject.h $DESTDIR/include/python3
install -m 644 Include/methodobject.h $DESTDIR/include/python3
install -m 644 Include/modsupport.h $DESTDIR/include/python3
install -m 644 Include/moduleobject.h $DESTDIR/include/python3
install -m 644 Include/namespaceobject.h $DESTDIR/include/python3
install -m 644 Include/object.h $DESTDIR/include/python3
install -m 644 Include/objimpl.h $DESTDIR/include/python3
install -m 644 Include/odictobject.h $DESTDIR/include/python3
install -m 644 Include/osmodule.h $DESTDIR/include/python3
install -m 644 Include/patchlevel.h $DESTDIR/include/python3
install -m 644 Include/pyarena.h $DESTDIR/include/python3
install -m 644 Include/pyatomic.h $DESTDIR/include/python3
install -m 644 Include/pycapsule.h $DESTDIR/include/python3
install -m 644 Include/pyctype.h $DESTDIR/include/python3
install -m 644 Include/pydebug.h $DESTDIR/include/python3
install -m 644 Include/pyerrors.h $DESTDIR/include/python3
install -m 644 Include/pyfpe.h $DESTDIR/include/python3
install -m 644 Include/pyhash.h $DESTDIR/include/python3
install -m 644 Include/pylifecycle.h $DESTDIR/include/python3
install -m 644 Include/pymacconfig.h $DESTDIR/include/python3
install -m 644 Include/pymacro.h $DESTDIR/include/python3
install -m 644 Include/pymath.h $DESTDIR/include/python3
install -m 644 Include/pymem.h $DESTDIR/include/python3
install -m 644 Include/pyport.h $DESTDIR/include/python3
install -m 644 Include/pystate.h $DESTDIR/include/python3
install -m 644 Include/pystrcmp.h $DESTDIR/include/python3
install -m 644 Include/pystrtod.h $DESTDIR/include/python3
install -m 644 Include/Python.h $DESTDIR/include/python3
install -m 644 Include/pythonrun.h $DESTDIR/include/python3
install -m 644 Include/pythread.h $DESTDIR/include/python3
install -m 644 Include/pytime.h $DESTDIR/include/python3
install -m 644 Include/rangeobject.h $DESTDIR/include/python3
install -m 644 Include/setobject.h $DESTDIR/include/python3
install -m 644 Include/sliceobject.h $DESTDIR/include/python3
install -m 644 Include/structmember.h $DESTDIR/include/python3
install -m 644 Include/structseq.h $DESTDIR/include/python3
install -m 644 Include/sysmodule.h $DESTDIR/include/python3
install -m 644 Include/traceback.h $DESTDIR/include/python3
install -m 644 Include/tupleobject.h $DESTDIR/include/python3
install -m 644 Include/typeslots.h $DESTDIR/include/python3
install -m 644 Include/unicodeobject.h $DESTDIR/include/python3
install -m 644 Include/warnings.h $DESTDIR/include/python3
install -m 644 Include/weakrefobject.h $DESTDIR/include/python3
ln -sf python3 $DESTDIR/include/python3.7
install -d $DESTDIR/lib
install libpython3.7.so.1 $DESTDIR/lib
ln -sf libpython3.7.so.1 $DESTDIR/lib/libpython3.7.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 Misc/python.pc $DESTDIR/lib/pkgconfig/python3.pc
install -d $DESTDIR/lib/python3
install -m 644 Lib/__future__.py $DESTDIR/lib/python3
install -m 644 Lib/_collections_abc.py $DESTDIR/lib/python3
install -m 644 Lib/_compat_pickle.py $DESTDIR/lib/python3
install -m 644 Lib/_compression.py $DESTDIR/lib/python3
install -m 644 Lib/_sitebuiltins.py $DESTDIR/lib/python3
install -m 644 Lib/_weakrefset.py $DESTDIR/lib/python3
install -m 644 Lib/abc.py $DESTDIR/lib/python3
install -m 644 Lib/argparse.py $DESTDIR/lib/python3
install -m 644 Lib/ast.py $DESTDIR/lib/python3
install -m 644 Lib/base64.py $DESTDIR/lib/python3
install -m 644 Lib/bdb.py $DESTDIR/lib/python3
install -m 644 Lib/bisect.py $DESTDIR/lib/python3
install -m 644 Lib/cProfile.py $DESTDIR/lib/python3
install -m 644 Lib/calendar.py $DESTDIR/lib/python3
install -m 644 Lib/cmd.py $DESTDIR/lib/python3
install -m 644 Lib/code.py $DESTDIR/lib/python3
install -m 644 Lib/codecs.py $DESTDIR/lib/python3
install -m 644 Lib/codeop.py $DESTDIR/lib/python3
install -m 644 Lib/configparser.py $DESTDIR/lib/python3
install -m 644 Lib/contextlib.py $DESTDIR/lib/python3
install -m 644 Lib/contextvars.py $DESTDIR/lib/python3
install -m 644 Lib/copy.py $DESTDIR/lib/python3
install -m 644 Lib/copyreg.py $DESTDIR/lib/python3
install -m 644 Lib/dataclasses.py $DESTDIR/lib/python3
install -m 644 Lib/datetime.py $DESTDIR/lib/python3
install -m 644 Lib/dis.py $DESTDIR/lib/python3
install -m 644 Lib/enum.py $DESTDIR/lib/python3
install -m 644 Lib/filecmp.py $DESTDIR/lib/python3
install -m 644 Lib/fnmatch.py $DESTDIR/lib/python3
install -m 644 Lib/functools.py $DESTDIR/lib/python3
install -m 644 Lib/genericpath.py $DESTDIR/lib/python3
install -m 644 Lib/gettext.py $DESTDIR/lib/python3
install -m 644 Lib/glob.py $DESTDIR/lib/python3
install -m 644 Lib/gzip.py $DESTDIR/lib/python3
install -m 644 Lib/hashlib.py $DESTDIR/lib/python3
install -m 644 Lib/heapq.py $DESTDIR/lib/python3
install -m 644 Lib/inspect.py $DESTDIR/lib/python3
install -m 644 Lib/io.py $DESTDIR/lib/python3
install -m 644 Lib/keyword.py $DESTDIR/lib/python3
install -m 644 Lib/linecache.py $DESTDIR/lib/python3
install -m 644 Lib/locale.py $DESTDIR/lib/python3
install -m 644 Lib/ntpath.py $DESTDIR/lib/python3
install -m 644 Lib/opcode.py $DESTDIR/lib/python3
install -m 644 Lib/operator.py $DESTDIR/lib/python3
install -m 644 Lib/optparse.py $DESTDIR/lib/python3
install -m 644 Lib/os.py $DESTDIR/lib/python3
install -m 644 Lib/pathlib.py $DESTDIR/lib/python3
install -m 644 Lib/pdb.py $DESTDIR/lib/python3
install -m 644 Lib/pickle.py $DESTDIR/lib/python3
install -m 644 Lib/pipes.py $DESTDIR/lib/python3
install -m 644 Lib/platform.py $DESTDIR/lib/python3
install -m 644 Lib/posixpath.py $DESTDIR/lib/python3
install -m 644 Lib/pprint.py $DESTDIR/lib/python3
install -m 644 Lib/profile.py $DESTDIR/lib/python3
install -m 644 Lib/queue.py $DESTDIR/lib/python3
install -m 644 Lib/quopri.py $DESTDIR/lib/python3
install -m 644 Lib/random.py $DESTDIR/lib/python3
install -m 644 Lib/re.py $DESTDIR/lib/python3
install -m 644 Lib/reprlib.py $DESTDIR/lib/python3
install -m 644 Lib/selectors.py $DESTDIR/lib/python3
install -m 644 Lib/shlex.py $DESTDIR/lib/python3
install -m 644 Lib/shutil.py $DESTDIR/lib/python3
install -m 644 Lib/signal.py $DESTDIR/lib/python3
install -m 644 Lib/site.py $DESTDIR/lib/python3
install -m 644 Lib/socket.py $DESTDIR/lib/python3
install -m 644 Lib/sre_compile.py $DESTDIR/lib/python3
install -m 644 Lib/sre_constants.py $DESTDIR/lib/python3
install -m 644 Lib/sre_parse.py $DESTDIR/lib/python3
install -m 644 Lib/stat.py $DESTDIR/lib/python3
install -m 644 Lib/string.py $DESTDIR/lib/python3
install -m 644 Lib/struct.py $DESTDIR/lib/python3
install -m 644 Lib/subprocess.py $DESTDIR/lib/python3
install -m 644 Lib/sysconfig.py $DESTDIR/lib/python3
install -m 644 Lib/tarfile.py $DESTDIR/lib/python3
install -m 644 Lib/tempfile.py $DESTDIR/lib/python3
install -m 644 Lib/textwrap.py $DESTDIR/lib/python3
install -m 644 Lib/threading.py $DESTDIR/lib/python3
install -m 644 Lib/token.py $DESTDIR/lib/python3
install -m 644 Lib/tokenize.py $DESTDIR/lib/python3
install -m 644 Lib/traceback.py $DESTDIR/lib/python3
install -m 644 Lib/types.py $DESTDIR/lib/python3
install -m 644 Lib/typing.py $DESTDIR/lib/python3
install -m 644 Lib/uu.py $DESTDIR/lib/python3
install -m 644 Lib/uuid.py $DESTDIR/lib/python3
install -m 644 Lib/warnings.py $DESTDIR/lib/python3
install -m 644 Lib/weakref.py $DESTDIR/lib/python3
install -m 644 Lib/zipfile.py $DESTDIR/lib/python3
install -d $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/__init__.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/base_events.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/base_futures.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/base_subprocess.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/base_tasks.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/constants.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/coroutines.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/events.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/format_helpers.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/futures.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/locks.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/log.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/protocols.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/queues.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/runners.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/selector_events.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/sslproto.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/streams.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/subprocess.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/tasks.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/transports.py $DESTDIR/lib/python3/asyncio
install -m 644 Lib/asyncio/unix_events.py $DESTDIR/lib/python3/asyncio
install -d $DESTDIR/lib/python3/collections
install -m 644 Lib/collections/__init__.py $DESTDIR/lib/python3/collections
install -m 644 Lib/collections/abc.py $DESTDIR/lib/python3/collections
install -d $DESTDIR/lib/python3/concurrent/futures
install -m 644 Lib/concurrent/futures/__init__.py $DESTDIR/lib/python3/concurrent/futures
install -m 644 Lib/concurrent/futures/_base.py $DESTDIR/lib/python3/concurrent/futures
install -m 644 Lib/concurrent/futures/thread.py $DESTDIR/lib/python3/concurrent/futures
install -d $DESTDIR/lib/python3/distutils
install -m 644 Lib/distutils/version.py $DESTDIR/lib/python3/distutils
install -d $DESTDIR/lib/python3/email
install -m 644 Lib/email/_encoded_words.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/_parseaddr.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/_policybase.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/base64mime.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/charset.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/encoders.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/errors.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/feedparser.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/header.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/iterators.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/message.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/parser.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/quoprimime.py $DESTDIR/lib/python3/email
install -m 644 Lib/email/utils.py $DESTDIR/lib/python3/email
install -d $DESTDIR/lib/python3/encodings
install -m 644 Lib/encodings/__init__.py $DESTDIR/lib/python3/encodings
install -m 644 Lib/encodings/aliases.py $DESTDIR/lib/python3/encodings
install -m 644 Lib/encodings/ascii.py $DESTDIR/lib/python3/encodings
install -m 644 Lib/encodings/latin_1.py $DESTDIR/lib/python3/encodings
install -m 644 Lib/encodings/unicode_escape.py $DESTDIR/lib/python3/encodings
install -m 644 Lib/encodings/utf_8.py $DESTDIR/lib/python3/encodings
install -d $DESTDIR/lib/python3/http
install -m 644 Lib/http/__init__.py $DESTDIR/lib/python3/http
install -m 644 Lib/http/client.py $DESTDIR/lib/python3/http
install -d $DESTDIR/lib/python3/importlib
install -m 644 Lib/importlib/__init__.py $DESTDIR/lib/python3/importlib
install -m 644 Lib/importlib/_bootstrap.py $DESTDIR/lib/python3/importlib
install -m 644 Lib/importlib/_bootstrap_external.py $DESTDIR/lib/python3/importlib
install -m 644 Lib/importlib/abc.py $DESTDIR/lib/python3/importlib
install -m 644 Lib/importlib/machinery.py $DESTDIR/lib/python3/importlib
install -m 644 Lib/importlib/resources.py $DESTDIR/lib/python3/importlib
install -m 644 Lib/importlib/util.py $DESTDIR/lib/python3/importlib
install -d $DESTDIR/lib/python3/json
install -m 644 Lib/json/__init__.py $DESTDIR/lib/python3/json
install -m 644 Lib/json/decoder.py $DESTDIR/lib/python3/json
install -m 644 Lib/json/encoder.py $DESTDIR/lib/python3/json
install -m 644 Lib/json/scanner.py $DESTDIR/lib/python3/json
install -d $DESTDIR/lib/python3/lib-dynload
install `cat pybuilddir.txt`/_blake2.* $DESTDIR/lib/python3/lib-dynload/_blake2.so
install `cat pybuilddir.txt`/_contextvars.* $DESTDIR/lib/python3/lib-dynload/_contextvars.so
install `cat pybuilddir.txt`/_lsprof.* $DESTDIR/lib/python3/lib-dynload/_lsprof.so
install `cat pybuilddir.txt`/_md5.* $DESTDIR/lib/python3/lib-dynload/_md5.so
install `cat pybuilddir.txt`/_posixsubprocess.* $DESTDIR/lib/python3/lib-dynload/_posixsubprocess.so
install `cat pybuilddir.txt`/_queue.* $DESTDIR/lib/python3/lib-dynload/_queue.so
install `cat pybuilddir.txt`/_random.* $DESTDIR/lib/python3/lib-dynload/_random.so
install `cat pybuilddir.txt`/_sha1.* $DESTDIR/lib/python3/lib-dynload/_sha1.so
install `cat pybuilddir.txt`/_sha256.* $DESTDIR/lib/python3/lib-dynload/_sha256.so
install `cat pybuilddir.txt`/_sha3.* $DESTDIR/lib/python3/lib-dynload/_sha3.so
install `cat pybuilddir.txt`/_sha512.* $DESTDIR/lib/python3/lib-dynload/_sha512.so
install `cat pybuilddir.txt`/_socket.* $DESTDIR/lib/python3/lib-dynload/_socket.so
install `cat pybuilddir.txt`/_struct.* $DESTDIR/lib/python3/lib-dynload/_struct.so
install `cat pybuilddir.txt`/array.* $DESTDIR/lib/python3/lib-dynload/array.so
install `cat pybuilddir.txt`/binascii.* $DESTDIR/lib/python3/lib-dynload/binascii.so
install `cat pybuilddir.txt`/fcntl.* $DESTDIR/lib/python3/lib-dynload/fcntl.so
install `cat pybuilddir.txt`/math.* $DESTDIR/lib/python3/lib-dynload/math.so
install `cat pybuilddir.txt`/pyexpat.* $DESTDIR/lib/python3/lib-dynload/pyexpat.so
install `cat pybuilddir.txt`/select.* $DESTDIR/lib/python3/lib-dynload/select.so
install `cat pybuilddir.txt`/unicodedata.* $DESTDIR/lib/python3/lib-dynload/unicodedata.so
install `cat pybuilddir.txt`/zlib.* $DESTDIR/lib/python3/lib-dynload/zlib.so
install -d $DESTDIR/lib/python3/logging
install -m 644 Lib/logging/__init__.py $DESTDIR/lib/python3/logging
install -d $DESTDIR/lib/python3/multiprocessing
install -m 644 Lib/multiprocessing/__init__.py $DESTDIR/lib/python3/multiprocessing
install -m 644 Lib/multiprocessing/context.py $DESTDIR/lib/python3/multiprocessing
install -m 644 Lib/multiprocessing/process.py $DESTDIR/lib/python3/multiprocessing
install -m 644 Lib/multiprocessing/reduction.py $DESTDIR/lib/python3/multiprocessing
install -d $DESTDIR/lib/python3/urllib
install -m 644 Lib/urllib/error.py $DESTDIR/lib/python3/urllib
install -m 644 Lib/urllib/parse.py $DESTDIR/lib/python3/urllib
install -m 644 Lib/urllib/request.py $DESTDIR/lib/python3/urllib
install -m 644 Lib/urllib/response.py $DESTDIR/lib/python3/urllib
install -d $DESTDIR/lib/python3/xml/etree
install -m 644 Lib/xml/etree/ElementPath.py $DESTDIR/lib/python3/xml/etree
install -m 644 Lib/xml/etree/ElementTree.py $DESTDIR/lib/python3/xml/etree
install -d $DESTDIR/lib/python3/xml/parsers
install -m 644 Lib/xml/parsers/expat.py $DESTDIR/lib/python3/xml/parsers
install -d $DESTDIR/lib/python3/xml/sax
install -m 644 Lib/xml/sax/__init__.py $DESTDIR/lib/python3/xml/sax
install -m 644 Lib/xml/sax/_exceptions.py $DESTDIR/lib/python3/xml/sax
install -m 644 Lib/xml/sax/expatreader.py $DESTDIR/lib/python3/xml/sax
install -m 644 Lib/xml/sax/handler.py $DESTDIR/lib/python3/xml/sax
install -m 644 Lib/xml/sax/saxutils.py $DESTDIR/lib/python3/xml/sax
install -m 644 Lib/xml/sax/xmlreader.py $DESTDIR/lib/python3/xml/sax
ln -sf python3 /lib/python3.7

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/python3-build.pc << EOF
Name: Python
Version: 3.7.17
Description: Python programming language
Requires: $REQUIRES

devel=\\
/include/python3/abstract.h \\
/include/python3/bltinmodule.h \\
/include/python3/boolobject.h \\
/include/python3/bytearrayobject.h \\
/include/python3/bytesobject.h \\
/include/python3/cellobject.h \\
/include/python3/ceval.h \\
/include/python3/classobject.h \\
/include/python3/code.h \\
/include/python3/codecs.h \\
/include/python3/compile.h \\
/include/python3/complexobject.h \\
/include/python3/context.h \\
/include/python3/descrobject.h \\
/include/python3/dictobject.h \\
/include/python3/dtoa.h \\
/include/python3/enumobject.h \\
/include/python3/eval.h \\
/include/python3/fileobject.h \\
/include/python3/fileutils.h \\
/include/python3/floatobject.h \\
/include/python3/frameobject.h \\
/include/python3/funcobject.h \\
/include/python3/genobject.h \\
/include/python3/import.h \\
/include/python3/intrcheck.h \\
/include/python3/iterobject.h \\
/include/python3/listobject.h \\
/include/python3/longintrepr.h \\
/include/python3/longobject.h \\
/include/python3/memoryobject.h \\
/include/python3/methodobject.h \\
/include/python3/modsupport.h \\
/include/python3/moduleobject.h \\
/include/python3/namespaceobject.h \\
/include/python3/object.h \\
/include/python3/objimpl.h \\
/include/python3/odictobject.h \\
/include/python3/osmodule.h \\
/include/python3/patchlevel.h \\
/include/python3/pyarena.h \\
/include/python3/pyatomic.h \\
/include/python3/pycapsule.h \\
/include/python3/pyconfig.h \\
/include/python3/pyctype.h \\
/include/python3/pydebug.h \\
/include/python3/pyerrors.h \\
/include/python3/pyfpe.h \\
/include/python3/pyhash.h \\
/include/python3/pylifecycle.h \\
/include/python3/pymacconfig.h \\
/include/python3/pymacro.h \\
/include/python3/pymath.h \\
/include/python3/pymem.h \\
/include/python3/pyport.h \\
/include/python3/pystate.h \\
/include/python3/pystrcmp.h \\
/include/python3/pystrtod.h \\
/include/python3/Python.h \\
/include/python3/pythonrun.h \\
/include/python3/pythread.h \\
/include/python3/pytime.h \\
/include/python3/rangeobject.h \\
/include/python3/setobject.h \\
/include/python3/sliceobject.h \\
/include/python3/structmember.h \\
/include/python3/structseq.h \\
/include/python3/sysmodule.h \\
/include/python3/traceback.h \\
/include/python3/tupleobject.h \\
/include/python3/typeslots.h \\
/include/python3/unicodeobject.h \\
/include/python3/warnings.h \\
/include/python3/weakrefobject.h \\
/include/python3.7 \\
/lib/libpython3.7.so \\
/lib/pkgconfig/python3.pc

exec=\\
/bin/python3 \\
/lib/libpython3.7.so.1 \\
/lib/python3/__future__.py \\
/lib/python3/_collections_abc.py \\
/lib/python3/_compat_pickle.py \\
/lib/python3/_compression.py \\
/lib/python3/_sitebuiltins.py \\
/lib/python3/_weakrefset.py \\
/lib/python3/abc.py \\
/lib/python3/argparse.py \\
/lib/python3/ast.py \\
/lib/python3/base64.py \\
/lib/python3/bdb.py \\
/lib/python3/bisect.py \\
/lib/python3/cProfile.py \\
/lib/python3/calendar.py \\
/lib/python3/cmd.py \\
/lib/python3/code.py \\
/lib/python3/codecs.py \\
/lib/python3/codeop.py \\
/lib/python3/configparser.py \\
/lib/python3/contextlib.py \\
/lib/python3/contextvars.py \\
/lib/python3/copy.py \\
/lib/python3/copyreg.py \\
/lib/python3/dataclasses.py \\
/lib/python3/datetime.py \\
/lib/python3/dis.py \\
/lib/python3/enum.py \\
/lib/python3/filecmp.py \\
/lib/python3/fnmatch.py \\
/lib/python3/functools.py \\
/lib/python3/genericpath.py \\
/lib/python3/gettext.py \\
/lib/python3/glob.py \\
/lib/python3/gzip.py \\
/lib/python3/hashlib.py \\
/lib/python3/heapq.py \\
/lib/python3/inspect.py \\
/lib/python3/io.py \\
/lib/python3/keyword.py \\
/lib/python3/linecache.py \\
/lib/python3/locale.py \\
/lib/python3/ntpath.py \\
/lib/python3/opcode.py \\
/lib/python3/operator.py \\
/lib/python3/optparse.py \\
/lib/python3/os.py \\
/lib/python3/pathlib.py \\
/lib/python3/pdb.py \\
/lib/python3/pickle.py \\
/lib/python3/pipes.py \\
/lib/python3/platform.py \\
/lib/python3/posixpath.py \\
/lib/python3/pprint.py \\
/lib/python3/profile.py \\
/lib/python3/queue.py \\
/lib/python3/quopri.py \\
/lib/python3/random.py \\
/lib/python3/re.py \\
/lib/python3/reprlib.py \\
/lib/python3/selectors.py \\
/lib/python3/shlex.py \\
/lib/python3/shutil.py \\
/lib/python3/signal.py \\
/lib/python3/site.py \\
/lib/python3/socket.py \\
/lib/python3/sre_compile.py \\
/lib/python3/sre_constants.py \\
/lib/python3/sre_parse.py \\
/lib/python3/stat.py \\
/lib/python3/string.py \\
/lib/python3/struct.py \\
/lib/python3/subprocess.py \\
/lib/python3/sysconfig.py \\
/lib/python3/tarfile.py \\
/lib/python3/tempfile.py \\
/lib/python3/textwrap.py \\
/lib/python3/threading.py \\
/lib/python3/token.py \\
/lib/python3/tokenize.py \\
/lib/python3/traceback.py \\
/lib/python3/types.py \\
/lib/python3/typing.py \\
/lib/python3/uu.py \\
/lib/python3/uuid.py \\
/lib/python3/warnings.py \\
/lib/python3/weakref.py \\
/lib/python3/zipfile.py \\
/lib/python3/asyncio/__init__.py \\
/lib/python3/asyncio/base_events.py \\
/lib/python3/asyncio/base_futures.py \\
/lib/python3/asyncio/base_subprocess.py \\
/lib/python3/asyncio/base_tasks.py \\
/lib/python3/asyncio/constants.py \\
/lib/python3/asyncio/coroutines.py \\
/lib/python3/asyncio/events.py \\
/lib/python3/asyncio/format_helpers.py \\
/lib/python3/asyncio/futures.py \\
/lib/python3/asyncio/locks.py \\
/lib/python3/asyncio/log.py \\
/lib/python3/asyncio/protocols.py \\
/lib/python3/asyncio/queues.py \\
/lib/python3/asyncio/runners.py \\
/lib/python3/asyncio/selector_events.py \\
/lib/python3/asyncio/sslproto.py \\
/lib/python3/asyncio/streams.py \\
/lib/python3/asyncio/subprocess.py \\
/lib/python3/asyncio/tasks.py \\
/lib/python3/asyncio/transports.py \\
/lib/python3/asyncio/unix_events.py \\
/lib/python3/collections/__init__.py \\
/lib/python3/collections/abc.py \\
/lib/python3/concurrent/futures/__init__.py \\
/lib/python3/concurrent/futures/_base.py \\
/lib/python3/concurrent/futures/thread.py \\
/lib/python3/distutils/version.py \\
/lib/python3/email/_encoded_words.py \\
/lib/python3/email/_parseaddr.py \\
/lib/python3/email/_policybase.py \\
/lib/python3/email/base64mime.py \\
/lib/python3/email/charset.py \\
/lib/python3/email/encoders.py \\
/lib/python3/email/errors.py \\
/lib/python3/email/feedparser.py \\
/lib/python3/email/header.py \\
/lib/python3/email/iterators.py \\
/lib/python3/email/message.py \\
/lib/python3/email/parser.py \\
/lib/python3/email/quoprimime.py \\
/lib/python3/email/utils.py \\
/lib/python3/encodings/__init__.py \\
/lib/python3/encodings/aliases.py \\
/lib/python3/encodings/ascii.py \\
/lib/python3/encodings/latin_1.py \\
/lib/python3/encodings/unicode_escape.py \\
/lib/python3/encodings/utf_8.py \\
/lib/python3/http/__init__.py \\
/lib/python3/http/client.py \\
/lib/python3/importlib/__init__.py \\
/lib/python3/importlib/_bootstrap.py \\
/lib/python3/importlib/_bootstrap_external.py \\
/lib/python3/importlib/abc.py \\
/lib/python3/importlib/machinery.py \\
/lib/python3/importlib/resources.py \\
/lib/python3/importlib/util.py \\
/lib/python3/json/__init__.py \\
/lib/python3/json/decoder.py \\
/lib/python3/json/encoder.py \\
/lib/python3/json/scanner.py \\
/lib/python3/lib-dynload/_blake2.so \\
/lib/python3/lib-dynload/_contextvars.so \\
/lib/python3/lib-dynload/_lsprof.so \\
/lib/python3/lib-dynload/_md5.so \\
/lib/python3/lib-dynload/_posixsubprocess.so \\
/lib/python3/lib-dynload/_queue.so \\
/lib/python3/lib-dynload/_random.so \\
/lib/python3/lib-dynload/_sha1.so \\
/lib/python3/lib-dynload/_sha256.so \\
/lib/python3/lib-dynload/_sha3.so \\
/lib/python3/lib-dynload/_sha512.so \\
/lib/python3/lib-dynload/_socket.so \\
/lib/python3/lib-dynload/_struct.so \\
/lib/python3/lib-dynload/array.so \\
/lib/python3/lib-dynload/binascii.so \\
/lib/python3/lib-dynload/fcntl.so \\
/lib/python3/lib-dynload/math.so \\
/lib/python3/lib-dynload/pyexpat.so \\
/lib/python3/lib-dynload/select.so \\
/lib/python3/lib-dynload/unicodedata.so \\
/lib/python3/lib-dynload/zlib.so \\
/lib/python3/logging/__init__.py \\
/lib/python3/multiprocessing/__init__.py \\
/lib/python3/multiprocessing/context.py \\
/lib/python3/multiprocessing/process.py \\
/lib/python3/multiprocessing/reduction.py \\
/lib/python3/urllib/error.py \\
/lib/python3/urllib/parse.py \\
/lib/python3/urllib/request.py \\
/lib/python3/urllib/response.py \\
/lib/python3/xml/etree/ElementPath.py \\
/lib/python3/xml/etree/ElementTree.py \\
/lib/python3/xml/parsers/expat.py \\
/lib/python3/xml/sax/__init__.py \\
/lib/python3/xml/sax/_exceptions.py \\
/lib/python3/xml/sax/expatreader.py \\
/lib/python3/xml/sax/handler.py \\
/lib/python3/xml/sax/saxutils.py \\
/lib/python3/xml/sax/xmlreader.py \\
/lib/python3.7

EOF

exit
--- Python-3.7.17.orig/configure.ac
+++ Python-3.7.17/configure.ac
@@ -119,7 +119,7 @@
 
 # Version number of Python's own shared library file.
 AC_SUBST(SOVERSION)
-SOVERSION=1.0
+SOVERSION=1
 
 # The later defininition of _XOPEN_SOURCE disables certain features
 # on Linux, so we need _GNU_SOURCE to re-enable them (makedev, tm_zone).
@@ -3443,7 +3443,6 @@
 then
     AC_DEFINE(WITH_PYMALLOC, 1,
      [Define if you want to compile in Python-specific mallocs])
-    ABIFLAGS="${ABIFLAGS}m"
 fi
 AC_MSG_RESULT($with_pymalloc)
 
@@ -4749,8 +4748,6 @@
 # * The Python implementation (always 'cpython-' for us)
 # * The major and minor version numbers
 # * --with-pydebug (adds a 'd')
-# * --with-pymalloc (adds a 'm')
-# * --with-wide-unicode (adds a 'u')
 #
 # Thus for example, Python 3.2 built with wide unicode, pydebug, and pymalloc,
 # would get a shared library ABI version tag of 'cpython-32dmu' and shared
